### ISSUE TITLE
names/identifiers in bash must be alphanumeric+underscore only (eg, no dash)

### ISSUE THREAD

**Issue by draeath** [koalaman/shellcheck#933]
shellcheck 0.3.7
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

It seems that shellcheck doesn't catch dashes used in identifier names (eg, functions). The allowable characters are documented [here](https://www.gnu.org/software/bash/manual/bashref.html#index-identifier):

> A `word` consisting solely of letters, numbers, and underscores, and beginning with a letter or underscore. `Name`s are used as shell variable and function names. Also referred to as an `identifier`.

For example, this does not return any warnings from `shellcheck -s bash` despite violating the above with `conda-activate` being an invalid name:

```
conda-activate()
{
    source /opt/"${USER}"/miniconda3/bin/activate "${@}";
}
```

**Comment by draeath** [comment]
Oh, I realize i didn't explicitly state it outside of the issue title, but this is specific to bash. I am unsure of any other shells, however.

**Comment by draeath** [comment]
Maybe duplicate of #529 (sorry!)

**Comment by rdebath** [comment]
Actually, that is a valid function name for bash, it might not be sensible one though.

This is a really silly one...
```
#!/bin/bash

x-./^%!:#,{}+=@~()
{
  echo running
}

echo defined

x-./^%!:#,{}+=@~

unset -f x-./^%!:#,{}+=@~

echo done
```

**Comment by gozdal** [comment]
The variables though shouldn't have dash in the name. Here is a script which might look okish to untrained eye:
```bash
#!/bin/bash

FOO-BAR=ala

echo "${FOO-BAR}"
```
When launched:
```
./x.sh: line 3: FOO-BAR=ala: command not found
BAR
```
